x-pack/metricbeat/module/statsd/server/registry.go

```diff
@@ -374,9 +374,9 @@ func (r *registry) GetOrNewSet(name string, tags map[string]string) *setMetric {
 }
 
 func (r *registry) metricHash(tags map[string]string) string {
-	b, err := json.Marshal(tags)
-	if err != nil { // shouldn't happen on a map[string]string
-		panic(err)
+	mapstrTags := common.MapStr{}
+	for k, v := range tags {
+		mapstrTags[k] = v
```

Comment:
Could remove the need for the loop being that a map[string]string should be able to be converted to a `common.MapStr`

`common.MapStr(tags.(map[string]interface{}))`

Reply:
unfortunately that's not possible. If I change `tags` to be an interface (required in order to do a type assertion in the first place) and then try to type assert a `map[string]string` to a `map[string]interface{}`: 
```
panic: interface conversion: interface {} is map[string]string, not map[string]interface {} [recovered]
```

I experimented with adding a hashing function that takes a `map[string]string` instead of a `MapStr` so it can be used directly, but the performance gain was minimal.